File: lista_de_exercicios/questao7/questao7.go
package main

type Professor struct {
    nome     string
    escolas  []*Escola
}

type Escola struct {
    nome       string
    professores []*Professor
}

func (p *Professor) AdicionarEscola(e *Escola) {
    p.escolas = append(p.escolas, e)
    e.AdicionarProfessor(p)
}

func (e *Escola) AdicionarProfessor(p *Professor) {
    for _, professor := range e.professores {
        if professor == p {
            return
        }
    }
    e.professores = append(e.professores, p)
}

func main() {}

File: lista_de_exercicios/questao7/questao7.py
class Professor:
    def __init__(self, nome):
        self.nome = nome
        self.escolas = []

    def adicionar_escola(self, escola):
        self.escolas.append(escola)
        escola.adicionar_professor(self)

class Escola:
    def __init__(self, nome):
        self.nome = nome
        self.professores = []

    def adicionar_professor(self, professor):
        if professor not in self.professores:
            self.professores.append(professor)
